ksqldb-engine/src/test/java/io/confluent/ksql/integration/SecureIntegrationTest.java

```diff
@@ -243,6 +254,196 @@ public void shouldWorkWithMinimalPrefixedAcls() {
+    ksqlConfig.put(KSQL_SERVICE_ID_CONFIG, serviceId);
+
+    givenAllowAcl(NORMAL_USER,
+        resource(CLUSTER, "kafka-cluster"),
+        ops(DESCRIBE_CONFIGS));
+
+    givenAllowAcl(NORMAL_USER,
+        resource(TOPIC, INPUT_TOPIC),
+        ops(READ));
+
+    givenAllowAcl(NORMAL_USER,
+        prefixedResource(TOPIC, prefix),
+        ops(ALL));
+
+    givenAllowAcl(NORMAL_USER,
+        resource(TOPIC, outputTopic),
+        ops(CREATE /* as the topic doesn't exist yet*/, WRITE));
+
+    givenAllowAcl(NORMAL_USER,
+        prefixedResource(GROUP, prefix),
+        ops(ALL));
+
+    givenTestSetupWithConfig(ksqlConfig);
+
+    TEST_HARNESS.getKafkaCluster().addUserAcl(
+        NORMAL_USER.username,
+        AclPermissionType.DENY,
+        resource(TOPIC, INPUT_TOPIC),
+        ops(READ)
+    );
+
+    // Then:
+    assertQueryFailsWithUserError(
+        String.format(
+            "CREATE STREAM %s AS SELECT * FROM %s;",
+            outputTopic,
+            INPUT_STREAM
+        ),
+        String.format(
+            "%s: Not authorized to access topics: [%s]",
+            TopicAuthorizationException.class.getName(),
+            INPUT_TOPIC
+        )
+    );
+  }
+
+  @Test
+  public void shouldClassifyGroupAuthorizationExceptionAsUserError() {
+    // Given:
+    final String serviceId = "my-service-id_";  // Defaults to "default_"
+    final String prefix = "_confluent-ksql-" + serviceId;
+
+    final Map<String, Object> ksqlConfig = getKsqlConfig(NORMAL_USER);
+    ksqlConfig.put(KSQL_SERVICE_ID_CONFIG, serviceId);
+
+    givenAllowAcl(NORMAL_USER,
+        resource(CLUSTER, "kafka-cluster"),
+        ops(DESCRIBE_CONFIGS));
+
+    givenAllowAcl(NORMAL_USER,
+        resource(TOPIC, INPUT_TOPIC),
+        ops(READ));
+
+    givenAllowAcl(NORMAL_USER,
+        resource(TOPIC, outputTopic),
+        ops(CREATE /* as the topic doesn't exist yet*/, WRITE));
+
+    givenAllowAcl(NORMAL_USER,
+        prefixedResource(TOPIC, prefix),
+        ops(ALL));
+
+    // Skip setting consumer group permissions
+
+    givenTestSetupWithConfig(ksqlConfig);
+
+    // Then:
+    assertQueryFailsWithUserError(
+        String.format(
+            "CREATE STREAM %s AS SELECT * FROM %s;",
+            outputTopic,
+            INPUT_STREAM
+        ),
+        String.format(
+            "%s: Not authorized to access group: %squery_",
+            GroupAuthorizationException.class.getName(),
+            prefix
+        ) + "%s"
+    );
+  }
+
+  @Test
+  public void shouldClassifyTransactionIdAuthorizationExceptionAsUserError() {
+    // Given:
+    final String serviceId = "my-service-id_";  // Defaults to "default_"
+    final String prefix = "_confluent-ksql-" + serviceId;
+
+    final Map<String, Object> ksqlConfig = getKsqlConfig(NORMAL_USER);
+    ksqlConfig.put(KSQL_SERVICE_ID_CONFIG, serviceId);
+    ksqlConfig.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.EXACTLY_ONCE_BETA);
```

Comment:
Why does this cause a transaction ID authorization exception? (Can we add a quick comment so readers know that's the purpose of setting this config in this test?)